test/models/git_repository_test.rb

```diff
@@ -65,6 +66,41 @@
       repository.clone!
       repository.commit_from_ref('master').must_match /^[0-9a-f]{7}$/
     end
+
+    it 'returns the full commit id with nil length' do
+      create_repo_with_tags
+      repository.clone!
+      repository.commit_from_ref('master', length: nil).must_match /^[0-9a-f]{40}$/
+    end
+
+    it 'returns nil if ref does not exist' do
```

Comment:
a `does not allow script insertion` with `ref = 'head" ; do something evil'` could be nice